sdl2promela/promela/model.py

```diff
@@ -32,12 +150,40 @@ def __init__(self):
         self.name = None
 
 
+class ArrayAccess(Expression):
+    """Access to element of array"""
+
+    variable: Union[VariableReference, "MtypeAccess"]
+    """Reference to array variable"""
+
+    index: Expression
+    """Expression to calculate index in the array"""
+
+    def __init__(self):
+        self.variable = None
+        self.index = None
+
+
+class MtypeAccess(Expression):
+    """Access to the field of mtype"""
+
+    mtype: Union["MtypeAccess", ArrayAccess, VariableReference]
```

Comment:
Isn't mtype an enum? Isn't this access to a named/scoped mtype? And why does it share the logic with ArrayAccess, as well as potentially variable reference (is that how utypes are handled). I believe the naming could be improved to avoid confusion.

Reply:
Oh my,,, It should be UtypeAccess. However, MemberAccess is better name

Reply:
Changed to MemberAccess